frame/composable-support/src/validation.rs

```diff
@@ -33,64 +95,77 @@ pub struct Valid;
 #[derive(Debug, Eq, PartialEq, Default)]
 pub struct Invalid;
 
-impl<T> Validate<Invalid> for T {
+impl<T> Validate<T, Invalid> for Invalid {
```

Comment:
would this work `impl<T> Validate<T> for Invalid`?

Reply:
No, we need to supply the `Validate` trait with two generic arguments. Otherwise, we couldn't support multiple flat/nested validations